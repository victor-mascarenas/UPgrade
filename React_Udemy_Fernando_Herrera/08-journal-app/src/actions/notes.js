import { type } from "@testing-library/user-event/dist/type";
import { db } from "../firebase/firebase-config";
import { types } from "../reducers/types/types";

export const startNewNote = () => {
    return async (dispatch, getState) => {
        const uid = getState().auth.uid;
        const newNote = {
            title: '',
            body: '',
            date: new Date().getTime()
        };
        const docRef = await db.collection(`${uid}/journal/notes`).add(newNote);
        dispatch(activeNote(docRef.id, newNote));
    };
};

export const activeNote = (id, note) => ({
    type: types.NOTES_ACTIVE,
    payload: {
        id,
        ...note
    }
});