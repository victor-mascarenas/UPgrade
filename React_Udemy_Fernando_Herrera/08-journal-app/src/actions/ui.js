import { types } from "../reducers/types/types";

export const setError = (msg) => {
    return ({
        type: types.UI_SET_ERROR,
        payload: {
            errorMsg: msg
        }
    });
};

export const removeError = () => {
    return ({
        type: types.UI_REMOVE_ERROR
    });
};