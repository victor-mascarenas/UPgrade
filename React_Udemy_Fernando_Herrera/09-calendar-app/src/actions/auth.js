import Swal from "sweetalert2";
import { fetchWithoutToken } from "../helpers/fetch";
import { types } from "../types/types";

const storeToken = (token) => {
    localStorage.setItem('token', token);
    localStorage.setItem('token-start', new Date().getTime());
};

export const startLogin = (email, password) => {
    return async (dispatch) => {
        const res = await fetchWithoutToken('auth', {
            email,
            password
        }, 'POST');
        const body = await res.json();
        if (body.ok) {
            storeToken(body.token);
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        } else {
            Swal.fire('Error', body.msg, 'error');
        }
    };
};
export const startRegister = (email, password, name) => {
    return async (dispatch) => {
        const res = await fetchWithoutToken('auth/new', {
            name,
            email,
            password
        }, 'POST');
        const body = await res.json();
        if (body.ok) {
            storeToken(body.token);
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        } else {
            Swal.fire('Error', body.msg, 'error');
        }
    };
};

const login = (user) => ({
    type: types.AUTH_LOGIN,
    payload: user
});