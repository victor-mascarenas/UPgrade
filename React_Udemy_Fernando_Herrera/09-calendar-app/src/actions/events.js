import Swal from "sweetalert2";
import { fetchWithToken } from "../helpers/fetch";
import { prepareEvents } from "../helpers/prepareEvents";
import { types } from "../types/types"

const eventAddNew = (event) => ({
    type: types.EVENT_ADD_NEW,
    payload: event
});
const eventLoaded = (events) => ({
    type: types.EVENT_LOADED,
    payload: events
});

export const eventStartAddNew = (event) => {
    return async (dispatch, getState) => {
        const {uid, name} = getState().auth;
        try {
            const resp = await fetchWithToken('events', event, 'POST');
            const body = await resp.json();
            if (body.ok) {
                event.id = body.evento.id;
                event.user = {
                    _id: uid,
                    name
                };
                dispatch(eventAddNew(event));
            } else {
                Swal.fire('Error', body.msg, 'error');
            }
        } catch(error) {
            Swal.fire('Error', error.message, 'error');
        }
    };
};
export const eventStartLoading = () => {
    return async (dispatch) => {
        try {
            const res = await fetchWithToken('events');
            const body = await res.json();
            if (body.ok) {
                const events = prepareEvents(body.events);
                dispatch(eventLoaded(events));
            } else {
                Swal.fire('Error', body.msg, 'error');
            }
        } catch (error) {
            Swal.fire('Error', error.message, 'error');
        }
    };
};

export const eventSetActive = (event) => ({
    type: types.EVENT_SET_ACTIVE,
    payload: event
});
export const eventClearActive = () => ({
    type: types.EVENT_CLEAR_ACTIVE
});
export const eventUpdated = (event) => ({
    type: types.EVENT_UPDATED,
    payload: event
});
export const eventDeleted = () => ({
    type: types.EVENT_DELETED
});
export const eventSlotSelected = (start, end) => ({
    type: types.EVENT_SLOT_SELECTED,
    payload: {
        start,
        end
    }
});