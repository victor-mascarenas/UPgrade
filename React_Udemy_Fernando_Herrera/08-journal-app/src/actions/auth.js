import { types } from "../reducers/types/types";
import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { endLoading, startLoading } from "./ui";
import Swal from 'sweetalert2';
import { notesLogout } from "./notes";

export const startLogin = (email, password) => {
    return (dispatch) => {//dispatch comes from Thunk
        dispatch(startLoading());
        return firebase.auth().signInWithEmailAndPassword(email, password)
            .then(({user}) => {
                dispatch(login(user.uid, user.displayName));
            })
            .catch(error => {
                Swal.fire('Error', error.message, 'error');
            })
            .finally(() => {
                dispatch(endLoading());
            });
    };
};

export const startCredentialsRegistration = (email, password, name) => {
    return (dispatch) => {
        firebase.auth().createUserWithEmailAndPassword(email, password)
            .then(async ({user}) => {
                await user.updateProfile({
                    displayName: name
                });
                dispatch(login(user.uid, user.displayName));
            })
            .catch(error => {
                Swal.fire('Error', error.message, 'error');
            });
    };
};

export const startGoogleLogin = () => {
    return (dispatch) => {
        firebase.auth().signInWithPopup(googleAuthProvider)
            .then(({user}) => {
                dispatch(login(user.uid, user.displayName));
            })
            .catch((error) => {
                Swal.fire('Error', error.message, 'error');
            });
    };
};

export const login = (uid, displayName) => (
    {
        type: types.LOGIN,
        payload: {
            uid,
            displayName
        }
    }
);

export const startLogout = () => {
    return async (dispatch) => {
        await firebase.auth().signOut();
        dispatch(logout());
        dispatch(notesLogout());
    };
};

export const logout = () => {
    return {
        type: types.LOGOUT
    };
}