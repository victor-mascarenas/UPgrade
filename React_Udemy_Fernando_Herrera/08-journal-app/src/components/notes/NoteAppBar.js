import React from 'react'
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNote, startUploading } from '../../actions/notes';

export const NoteAppBar = () => {
    const dispatch = useDispatch();
    const {active: note} = useSelector(state => state.notes);

    const fileInputStyle = {
        display: 'none'
    };

    const handleSave = () => {
        dispatch(startSaveNote(note));
    };
    const handlePicture = () => {
        document.querySelector('#fileSelector').click();
    };
    const handleFileChanged = (e) => {
        const file = e.target.files[0];
        if (file) {
            dispatch(startUploading(file));
        }
    };

    return (
        <div className='notes__appbar'>
            <span>28 de Agosto de 2022</span>
            <input id='fileSelector' name='file' type='file' style={fileInputStyle} onChange={handleFileChanged}/>
            <div>
                <button className='btn' onClick={handlePicture}>Add picture</button>
                <button className='btn' onClick={handleSave}>Save</button>
            </div>
        </div>
    )
}
