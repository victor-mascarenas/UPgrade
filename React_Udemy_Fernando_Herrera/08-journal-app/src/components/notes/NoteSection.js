import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { activeNote, startDeleting } from '../../actions/notes';
import useForm from '../../hooks/useForm';
import { NoteAppBar } from './NoteAppBar'

export const NoteSection = () => {
    const {active: note} = useSelector(state => state.notes);
    const {values: formValues, inputOnChange, reset} = useForm(note);
    const dispatch = useDispatch();

    useEffect(() => {
        reset();
    }, [note.id]);
    useEffect(() => {
        dispatch(activeNote(note.id, {...formValues}));
    }, [formValues]);

    const {body, title} = formValues;

    const handleDelete = () => {
        dispatch(startDeleting(note.id));
    };

    return (
        <div className='notes__main-content'>
            <NoteAppBar/>
            <div className='notes__content'>
                <input type='text' placeholder='Some awesome title' className='notes__title-input' name='title' value={title} onChange={inputOnChange}/>
                <textarea placeholder='What happened today?' className='notes__text-area' name='body' value={body} onChange={inputOnChange}></textarea>
                {
                    (note.url) &&
                        <div className='notes__images'>
                            <img src={note.url} alt='Cocina'/>
                        </div>
                }
            </div>
            <button className='btn btn-danger' onClick={handleDelete}>Delete</button>
        </div>
    )
}
