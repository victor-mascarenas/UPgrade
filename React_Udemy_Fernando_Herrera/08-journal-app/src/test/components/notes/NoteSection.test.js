import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';
import { activeNote } from '../../../actions/notes';
import { NoteSection } from '../../../components/notes/NoteSection';

jest.mock('../../../actions/notes', () => ({
    activeNote: jest.fn()
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: ''
    },
    notes: {
        notes: [],
        active: {
            id: '1234',
            title: 'Hola',
            body: 'mundo',
            date: 0
        }
    }
};
const store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <NoteSection/>
        </MemoryRouter>
    </Provider>
);

describe('Pruebas en NoteSection', () => {
    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de disparar el activeNote', () => {
        const target = {
            name: 'title',
            value: 'Hola de nuevo'
        };
        wrapper.find('input[name="title"]').simulate('change', {
            target
        });
        expect(activeNote).toHaveBeenLastCalledWith('1234', {
            body: initState.notes.active.body,
            title: target.value,
            id: '1234',
            date: 0
        });
    });
});