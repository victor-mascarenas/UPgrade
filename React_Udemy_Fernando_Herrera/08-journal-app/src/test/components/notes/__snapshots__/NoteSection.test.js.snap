// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en NoteSection Debe de mostrarse correctamente 1`] = `
<div
  className="notes__main-content animate__animated animate__fadeIn"
>
  <div
    className="notes__appbar"
  >
    <span>
      28 de Agosto de 2022
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Add picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="Hola"
    />
    <textarea
      className="notes__text-area"
      name="body"
      onChange={[Function]}
      placeholder="What happened today?"
      value="mundo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
